Classes/EventStore/Storage/Doctrine/DoctrineEventStorage.php

```diff
@@ -147,6 +147,11 @@ public function commit(StreamName $streamName, WritableEvents $events, int $expe
         while (true) {
             $this->reconnectDatabaseConnection();
             $this->connection->beginTransaction();
+
+            # `SELECT FOR UPDATE` write-locks affected data, `SELECT MAX() FOR UPDATE` write-locks the table.
+            # This should be "LOCK TABLE" but "LOCK TABLE" doesn't play nice with transactions.
+            $this->connection->query('SELECT MAX(sequencenumber) FROM ' . $this->eventTableName . ' FOR UPDATE')->fetchAll();
```

Comment:
```suggestion
            $this->connection->executeQuery('SELECT MAX(sequencenumber) FROM ' . $this->eventTableName . ' FOR UPDATE');
```
Do we really need the fetchAll()?

Reply:
Oh I see, `SELECT` yields results, so it should be *fetch*